summary_workflows/quantification/quantification_dockers/q_metrics/compute_metrics.py

```diff
@@ -52,53 +54,55 @@ def compute_metrics(participant_input, gold_standards_dir, challenge_ids, partic
         genome = matchPAS.load_genome(genome_file)
 
         for window in windows:
+            for return_df_type in all_return_df_types:
 
-            # METRIC: Matched sites
-            ########################
-            # metric on the number of matched sites
-            match_with_gt_run = matchPAS.match_with_gt(participant_input,gold_standard,window)
-            merged_bed_df, expression_unmatched = match_with_gt_run[0], match_with_gt_run[1]
-
-            # Key: exact name of metric as it appears in specification
-            metric_name = f"Expression_non-matched-PAS_{window}nt"
-            # Value: List of [variable_holding_metric, std_err]
-            metrics[metric_name] = [expression_unmatched, 0]
-
-
-            # METRIC: correlation coefficient
-            #################################
-            # metric on correlation coefficient
-            correlation_pearson = matchPAS.corr_Pearson_with_gt(merged_bed_df)
-            # Key: exact name of metric as it appears in specification
-            metric_name = f"Correlation_coefficient_Pearson_{window}nt"
-            # Value: List of [variable_holding_metric, std_err]
-            metrics[metric_name] = [correlation_pearson, 0]
-            # Spearman correlation
-            correlation_spearman = matchPAS.corr_Spearman_with_gt(merged_bed_df)
-            # Key: exact name of metric as it appears in specification
-            metric_name = f"Correlation_coefficient_Spearman_{window}nt"
-            # Value: List of [variable_holding_metric, std_err]
-            metrics[metric_name] = [correlation_spearman, 0]
-
-
-            # METRIC: correlation coefficient of relative pas usage
-            ####################
-            # Only calculate metric for first window size.
-            if window == windows[0]:
-                match_with_gt_and_pd = matchPAS.match_with_gt(participant_input,gold_standard,window, "with_unmatched_GT_and_PD")
-                merged_bed_df_wpd, _ = match_with_gt_and_pd[0], match_with_gt_and_pd[1]
-                normalised_df = matchPAS.relative_pas_usage(merged_bed_df_wpd, genome)
-                correlation_Pearson_rel_use = matchPAS.corr_Pearson_with_gt(normalised_df)
+                # METRIC: Matched sites
+                ########################
+                # metric on the number of matched sites
+                match_with_gt_run = matchPAS.match_with_gt(f_PD=participant_input, f_GT=gold_standard, 
```

Comment:
With the refactoring you introduced (returning the df here and the passing it to the Pearson and Spearman functions), this call should actually go above "METRIC: Matched sites". The return of `match_with_gt` is used in the three subsequent metric calculations